storage_controller/src/http.rs

```diff
@@ -749,6 +750,55 @@ impl From<ReconcileError> for ApiError {
     }
 }
 
+/// Return the safekeeper record by instance id, or 404.
+///
+/// Not used by anything except manual testing.
+async fn handle_get_safekeeper(req: Request<Body>) -> Result<Response<Body>, ApiError> {
+    // TODO: jwt, see POST handler in handle_upsert_safekeeper
+    let id = parse_request_param::<i64>(&req, "id")?;
+
+    let state = get_state(&req);
+
+    let res = state.service.get_safekeeper(id).await;
+
+    match res {
+        Ok(b) => json_response(StatusCode::OK, b),
+        Err(crate::persistence::DatabaseError::Query(diesel::result::Error::NotFound)) => {
+            Err(ApiError::NotFound("unknown instance_id".into()))
+        }
+        Err(other) => Err(other.into()),
+    }
+}
+
+/// Used as part of deployment scripts.
+///
+/// Assumes information is only relayed to storage controller after first selecting an unique id on
+/// control plane database, which means we have an id field in the request and payload.
```

Comment:
Is the deployment script going to call this endpoint multiple times for any given safekeeper? If yes, then we will update the db tuple each time even though it hasn't changed. Perhaps we should check if it's already registered and return a 2XX response in that case?

Reply:
I think the answer is yes, but quite rarely. It's easy to see how this could be used more often, or less often, as you pointed out.

I wouldn't be worried about polluting the table with dead tuples just yet.